authproxy/cmd/serve.go

```diff
@@ -107,10 +97,22 @@ func serve(cmd *cobra.Command, args []string) {
 		},
 	}))
 
-	remoteURL, err := url.Parse(remote)
+	remoteURLStr := os.Getenv(remoteURLEnvKey)
+	if remoteURLStr == "" {
+		e.Logger.Fatalf("The '%s' environment variable is needed to start the authentication proxy", remoteURLEnvKey)
+	}
+	remoteURL, err := url.Parse(remoteURLStr)
 	if err != nil {
 		e.Logger.Fatal(err)
 	}
+	proxyPortStr := os.Getenv(authProxyPortEnvKey)
+	if proxyPortStr == "" {
+		proxyPortStr = "65535"
+	}
+	authProxyPort, err := strconv.Atoi(proxyPortStr)
+	if err != nil {
+		e.Logger.Fatalf("Cannot convert the proxy port environment variable to integer: %v", err)
+	}
```

Comment:
I am not sure I am following the why of this change.

Both the port and URL could already be passed through environment variables with the original implementation.

See the two BindEnv calls you removed. Also, in `root.go`, I set the prefix for the environment variables to `AUTHPROXY` so you can use `AUTHPROXY_REMOTE` and `AUTHPROXY_PORT`.

Reply:
@sgaist the main reason for this is that I do not want to allow people to modify the port and remote for the proxy. Because in order to do this effectively you need to know a lot of intricacies about how amalthea sessions are setup. And it is easy to misunderstand this or mess it up. 

Also IMO allowing this to be modifiable leaks implementation details that should be hidden behind the CRD. The new CRD gives 3 options when it comes to authentication which are already really flexible. I.e. no authentication, authentication via oauth2 or authentication with a cookie/header. If someone wants to do something fancier they can turn off the amalthea authentication, add their own fully custom containers and add their own custom ingress. But when users go for the authentication amalthea provides, we constrain things a little bit but we remove the guesswork of what the remote url and port should be in the oauth proxy configuration. We can already put this configuration together from the things that a user would provide in the CR.

Reply:
OK, I think I am starting to follow your trail of thoughts. I had something similar in mind and the current implementation already provides the behaviour you want based on the order of handling used by viper as described [here](https://github.com/spf13/viper#why-viper):
```
Viper uses the following precedence order. Each item takes precedence over the item below it:

    explicit call to Set
    flag
    env
    config
    key/value store
    default
```

Environment variables go above configuration file so even if a user tries to change a value there, the final say is in the operator setting the environment variables or the corresponding parameter. So we have best of both worlds: the command is still easily testable while the configuration can be overridden by the operator when necessary.
Note that not all settings currently have matching environment variables.

Reply:
Ok yeah it was not clear to me how viper loads configurations. That makes sense. So I put the flags about port and url I removed. And I set them in the children resources via environment variables.